### ISSUE TITLE
Can't add plugin iptables

### ISSUE THREAD

**Issue by anhnvname** [collectd/collectd#1940]
Hello,

When I add iptables plugin, I got this error

```
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Sep 17 10:23:14 serv34 collectd[5760]: Found a configuration for the `iptables' plugin, but the plugin isn't loaded or didn't register a configuration callback.
```

Anybody got this trouble and fixed ?

**Comment by rpv-tomsk** [comment]
Did your config has correct `LoadPlugin iptables` line?
What `ldd` output showson `iptables.so`? Is it ok?

I have:
```
srv04:/usr/lib/collectd# ldd ./iptables.so
        linux-gate.so.1 (0xb7764000)
        libip4tc.so.0 => /lib/libip4tc.so.0 (0xb774d000)
        libip6tc.so.0 => /lib/libip6tc.so.0 (0xb7744000)
        libiptc.so.0 => /lib/libiptc.so.0 (0xb7741000)
        libdl.so.2 => /lib/i386-linux-gnu/i686/nosegneg/libdl.so.2 (0xb773b000)
        libc.so.6 => /lib/i386-linux-gnu/i686/nosegneg/libc.so.6 (0xb758e000)
        /lib/ld-linux.so.2 (0xb7765000)
```

**Comment by anhnvname** [comment]
Thanks @rpv-tomsk seem i don't see iptables.so files
I will re-check, but i don't understand, when I compile collectd I have make all why iptables doesn't include ?

**Comment by anhnvname** [comment]
Thank @rpv-tomsk, I have install iptables-devel now and I see iptables.so but no metric iptables sent

**Comment by rpv-tomsk** [comment]
What is your config?

**Comment by anhnvname** [comment]
Thanks for reply, Please check my config if i wrong

```
Interval     10
Timeout         10
LoadPlugin cpu
LoadPlugin df
LoadPlugin disk
LoadPlugin interface
LoadPlugin load
LoadPlugin memory
LoadPlugin swap
LoadPlugin uptime
LoadPlugin FileCount
LoadPlugin write_graphite
LoadPlugin iptables

<Plugin "iptables">
  Chain "filter" "INPUT"
  Chain "filter" "FORWARD"
  Chain "filter" "OUTPUT"
  Chain "filter" "ALLOWIN"
  Chain "filter" "ALLOWOUT"
  Chain "filter" "DENYIN"
  Chain "filter" "DENYOUT"
  Chain "filter" "INVALID"
  Chain "filter" "INVDROP"
  Chain "filter" "LOCALINPUT"
  Chain "filter" "LOCALOUTPUT"
  Chain "filter" "LOGDROPIN"
  Chain "filter" "LOGDROPOUT"
</Plugin>
<Plugin "disk">
  Disk "/^sd/"
  Disk "/^xvd/"
  IgnoreSelected false
</Plugin>
<Plugin "interface">
  Interface "/^bond/"
  Interface "/^eth/"
  Interface "/^em/"
  Interface "/^ens/"
  IgnoreSelected false
</Plugin>
<Plugin "filecount">
  <Directory "/var/spool/exim/input">
    Instance "mail-queue"
  </Directory>
</Plugin>
<Plugin "write_graphite">
 <Carbon>
   Host "10.5.5.5"
   Port "1995"
   Prefix "company.server01."
   Protocol "tcp"
   EscapeCharacter "_"
   SeparateInstances false
   StoreRates true
   AlwaysAppendDS false
 </Carbon>
</Plugin>

```

**Comment by rpv-tomsk** [comment]
I see no problems in config.

**Comment by octo** [comment]
@anhnvname If you compiled *collectd* yourself, you probably don't have the required library installed or are at least missing the development package of the library. There should be a line in the logs informing you that loading the *IPTables plugin* failed (and why).

If at all possible, I'd recommend to use a binary package for your distribution – https://pkg.ci.collectd.org/ has up-to-date packages for Debian / Ubuntu and RedHat.

**Comment by anhnvname** [comment]
thanks reply from @octo but I have installed iptables-devel on CentOS 7.
I'm passed error from beginning.
Now i don't see any metric to Graphite

**Comment by octo** [comment]
@anhnvname it still sounds like you're missing a dependency. Can you please provide the `config.log` written by the `configure` script?

**Comment by anhnvname** [comment]
Thanks @octo Please check my config.log follow URL
https://paste.ubuntu.com/23206435/

**Comment by octo** [comment]
@anhnvname The configure script complains that it can't find the `iptc_handle_t` type:

```
configure:25464: checking for iptc_handle_t
configure:25464: gcc -std=gnu99 -c -g -O2    conftest.c >&5
conftest.c: In function 'main':
conftest.c:225: error: 'iptc_handle_t' undeclared (first use in this function)
conftest.c:225: error: (Each undeclared identifier is reported only once
conftest.c:225: error: for each function it appears in.)
configure:25464: $? = 1
```

Which version of the `iptables-devel` do you have installed?

**Comment by anhnvname** [comment]
Hello,

I'm, the version of `iptables-devel` is `iptables-devel-1.4.7-16.el6.x86_64`

**Comment by anhnvname** [comment]
Hello @octo ,
Do you have any update for me

Thanks you

**Comment by rpv-tomsk** [comment]
I have the following on my Debian system:

```
# dpkg-query -l |grep iptables
ii  iptables                              1.4.14-3.1                   administration tools for packet filtering and NAT
ii  iptables-dev                          1.4.14-3.1                   iptables development files
```

```
./configure
...
checking libiptc/libiptc.h usability... yes
checking libiptc/libiptc.h presence... yes
checking for libiptc/libiptc.h... yes
checking libiptc/libip6tc.h usability... yes
checking libiptc/libip6tc.h presence... yes
checking for libiptc/libip6tc.h... yes
checking for iptc_handle_t... no
checking for ip6tc_handle_t... no
checking for library containing iptc_init... -liptc
...

 Modules:
...
    iptables  . . . . . . yes
```

part of `config.log`:

```
configure:25477: checking for iptc_handle_t
configure:25477: gcc -std=gnu99 -c -g -O2    conftest.c >&5
conftest.c: In function 'main':
conftest.c:233: error: 'iptc_handle_t' undeclared (first use in this function)
conftest.c:233: error: (Each undeclared identifier is reported only once
conftest.c:233: error: for each function it appears in.)
configure:25477: $? = 1
configure: failed program was:
```

The `iptc_handle_t` is missing but module compiles successfully. 
Missing `iptc_handle_t` is no problem, I think.

**Comment by anhnvname** [comment]
@rpv-tomsk . I think so. Iptables now complie correctly but it's doesn't send any metric

**Comment by rubenk** [comment]
Sorry, I can't reproduce this. As @octo says, try the packages for your distribution from https://pkg.ci.collectd.org/. If these don't work either, please open a new issue with as much detail as you can gather.
